### PR TITLE
feat(taskframework): JobCaller to operate odc task for different deploy sense 

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

module-task framework
type-feature

#### What this PR does / why we need it:

1.   add interface `JobCaller` for  operate odc job for different deployment environment, eg: k8s, master-worker or same jvm process
2. `K8sJobCaller` is used to operate odc task which running in a k8s pod
3. `JobContext` is odc job execution context contains all data for task running, and it's be set as pod environment
4. `JobCallerListener` will listen job call action and received notification from event publish
5. `TaskConnectionConfig` represent  connection configuration for task
6.  `TaskParam` contain task parameter which user submitted

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/task/context/JobContext.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     */
+    private Long taskId;
+    /**
+     * task param
+     */
+    private TaskParam taskParam;
+    /**
+     * task connection config
+     */
+    private TaskConnectionConfig connConfig;
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/task/context/JobContext.java:44]: if connection id is null in task entity,  we set null in connConfig

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/BaseJobCaller.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import com.oceanbase.odc.service.task.listener.JobCallerEvent;
+
+/**
+ * @author yaobin
+ * @date 2023-11-16
+ * @since 4.2.4
+ */
+public abstract class BaseJobCaller implements JobCaller {
+
+    private final EventPublisher publisher = new LocalEventPublisher();
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/BaseJobCaller.java:32]: listener can be add in  TaskFrameworkConfiguration see next pr

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/JobCallerEvent.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * @date 2023-11-16
+ * @since 4.2.4
+ */
+public class JobCallerEvent extends AbstractEvent {
+
+    @Getter
+    private final JobCallerAction jobAction;
+
+    @Getter
+    private final boolean status;
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/JobCallerEvent.java:33]: good, i suggest `successful`

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**